iris-location-service/src/main/java/iris/location_service/jsonrpc/LocationRPCImpl.java

```diff
@@ -29,90 +35,176 @@
+	private static final String EXCEPTION_MESSAGE_PROVIDER_ID = " - providerId: ";
 	private final @NotNull LocationService locationService;
 	private final @NotNull HealthEndpoint healthEndpoint;
 
 	@Override
 	public String postLocationsToSearchIndex(JsonRpcClientDto client, List<LocationInformation> locationList) {
 
-		var listOfInvalidAddresses = locationService.addLocations(client.getName(), locationList);
+		if (!isJsonRpcClientDtoInputValid(client)) {
+			throw new ResponseStatusException(HttpStatus.BAD_REQUEST, ErrorMessageHelper.INVALID_INPUT_EXCEPTION_MESSAGE);
+		}
+
+		List<LocationInformation> locationListValidated = locationInformationInputValidated(locationList);
+
+		var listOfInvalidAddresses = locationService.addLocations(client.getName(), locationListValidated);
 
 		var ret = "OK";
 		var logRes = ret;
 
 		if (!listOfInvalidAddresses.isEmpty()) {
 
-			ret = listOfInvalidAddresses.stream()
-					.collect(Collectors.joining(", ", "Invalid Locations detected: ", ""));
+			ret = listOfInvalidAddresses.stream().collect(Collectors.joining(", ", "Invalid Locations detected: ", ""));
 			logRes = "Invalid Locations detected";
 		}
 
-		log.debug("JSON-RPC - Post locations for client: {} (locations: {}) => result: {}", client.getName(),
-				locationList.size(), logRes);
+		log.debug("JSON-RPC - Post locations for client: {} (locations: {}) => result: {}", client.getName(), locationList.size(), logRes);
 
 		return ret;
 	}
 
 	@Override
 	public List<LocationOverviewDto> getProviderLocations(JsonRpcClientDto client) {
 
+		validateInputForGetProviderLocations(client);
+
 		var providerLocations = locationService.getProviderLocations(client.getName());
 
-		log.debug("JSON-RPC - Get provider locations for client: {} => returns {} locations", client.getName(),
-				providerLocations.size());
+		log.debug("JSON-RPC - Get provider locations for client: {} => returns {} locations", client.getName(), providerLocations.size());
 
 		return providerLocations;
 	}
 
 	@Override
 	public String deleteLocationFromSearchIndex(JsonRpcClientDto client, String locationId) {
+		validateInputForDeleteLocationFromSearchIndex(client, locationId);
 
 		var ret = "NOT FOUND";
 
 		if (locationService.deleteLocation(client.getName(), locationId))
 			ret = "OK";
 
-		log.debug("JSON-RPC - Delete location for client: {}; locationId: {} => result: {}", client.getName(),
-				obfuscate(locationId), ret);
+		log.debug("JSON-RPC - Delete location for client: {}; locationId: {} => result: {}", client.getName(), obfuscate(locationId), ret);
 
 		return ret;
 	}
 
 	@Override
 	public LocationQueryResult searchForLocation(JsonRpcClientDto client, String searchKeyword, PageableDto dto) {
 
-		var pageable = PageRequest.of(dto.getPage(), dto.getSize(),
-				dto.getSortBy() != null
-						? Sort.by(dto.getDirection(), dto.getSortBy())
-						: Sort.unsorted());
+		if (searchKeyword != null && !ValidationHelper.isNotShowingSignsForAttacks(searchKeyword)) {
```

Comment:
Actually this is meant to differentiate between required values and not required. It is meant to only make the valid check relevant if actually something is in the variable. While a null value here is acceptable as well.

Reply:
I find the solution with the two methods much better, because you have to think less about what the checks are for. We could still argue about whether the checks for attacks and mandatory fields should be linked, but I think it's ok here.